app/models/pawn.rb

```diff
@@ -1,6 +1,29 @@
 class Pawn < Piece
-  def move_valid?
-    # add logic here
+  def move_valid?(x, y)
+    return false if space_occupied?(x, y) == true
+    return false if legal_move?(x, y) == false
+    true
+  end
+
+  # basic one-step move:
+  def one_step_legal?(x, y)
+    return true if black_one_step_is_ok(x, y) || white_one_step_is_ok(x, y)
+    false
+  end
+
+  def black_one_step_is_ok(x, y)
+    return true if is_black? && (x_position == x && (y_position - y) == 1)
```

Comment:
`is_black?` with the question mark is actually an automatically-generated method that exists because the database column `is_black` exists, and it means "is there a value set for this column."  You probably meant to use `is_black` without the question mark.